cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -39,6 +39,46 @@
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
+    private URI addCookieInterceptorIfRequired(URI uri) {
+        String urlProtocol = uri.getScheme();
+        String urlHost = uri.getHost();
+        String urlPath = uri.getRawPath();
+
+        //Check if port exists
+        int urlPort = uri.getPort();
+        if (urlPort < 0) {
+            if (urlProtocol.equals("http")) {
```

Comment:
I'm pretty certain this can NPE. `uri.getScheme()` can return null if the `URI` has no scheme (see the [URI javadoc](http://developer.android.com/reference/java/net/URI.html#getScheme())), and since the URI comes from the library user's code rather than being something we're in control of, it could therefore have a null scheme.